src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -144,53 +139,29 @@ public RabbitMQQueueClient(ConnectionFactory connectionFactory, string queueName
             Func<IQueueMessage, CancellationToken, Task> messageHandler,
             MessageHandlerOptions messageHandlerOptions)
         {
+            lock (_channel)
+            {
+                // Reconfigure QoS in order to support specified level of concurrency.
+                // Here we are assuming a "worker queue", where multiple consumers are
+                // competing to draw from a single queue in order to spread workload
+                // across the consumers.
+                _channel.BasicQos(
+                    prefetchSize: 0,
+                    prefetchCount: Convert.ToUInt16(messageHandlerOptions.MaxConcurrentCalls),
+                    global: false);
+            }
+
             var consumer = new EventingBasicConsumer(_channel);
 
             consumer.Received += (channel, eventArgs) =>
             {
-                var cancellationToken = CancellationToken.None;
-
-                var message = new RabbitMQQueueMessage(
-                    eventArgs.Body,
-                    eventArgs.DeliveryTag,
-                    eventArgs.BasicProperties);
-
-                try
-                {
-                    messageHandler(message, cancellationToken)
-                        .GetAwaiter()
-                        .GetResult();
-
-                    if (messageHandlerOptions.AutoComplete)
-                    {
-                        CompleteAsync(message, cancellationToken)
-                            .GetAwaiter()
-                            .GetResult();
-                    }
-                }
-                catch (Exception ex)
-                {
-                    try
-                    {
-                        var exceptionHandler = messageHandlerOptions?.ExceptionReceivedHandler;
-
-                        if (exceptionHandler != null)
-                        {
-                            var context = new MessageHandlerExceptionContext(ex);
-
-                            exceptionHandler(context)
-                                .GetAwaiter()
-                                .GetResult();
-                        }
-                    }
-                    catch
-                    {
-                    }
-
-                    AbandonAsync(message, cancellationToken)
-                        .GetAwaiter()
-                        .GetResult();
-                }
+                // Handle the message asynchronously and do not await the result.
+                // This allows the `Received` event handler to continue and return
+                // before the message handler completes, which is essential for
+                // handling multiple messages concurrently with a single consumer.
+                // Any exceptions propagating from the message handler will
+                // effectively be ignored.
+                Task.Run(() => OnMessageReceived(messageHandler, messageHandlerOptions, eventArgs));
```

Comment:
In calling `Task.Run`, we're being explicit in our intention for the work to run on a separate thread. There are other ways of writing this code, e.g. using task continuations. The approach here was the simplest change to unblock the RabbitMQ consumer / channel thread, so that we can support handling multiple messages concurrently.